sdk/test/event/libevent/io_dispatcher_test.cc

```diff
@@ -0,0 +1,32 @@
+#include "src/event/libevent/io_dispatcher.h"
+
+#include <chrono>
+
+#include <gtest/gtest.h>
+using opentelemetry::sdk::event::Timer;
+using opentelemetry::sdk::event::libevent::IoDispatcher;
+
+TEST(DispatcherTest, Timer)
+{
+  IoDispatcher dispatcher;
+  std::chrono::steady_clock::time_point t1, t2;
+  auto timer = dispatcher.CreateTimer([&] { t2 = std::chrono::steady_clock::now(); });
+  timer->EnableTimer(std::chrono::milliseconds{100});
+  t1 = std::chrono::steady_clock::now();
+  dispatcher.Run();
+  auto duration = t2 - t1;
+  EXPECT_TRUE(duration > std::chrono::milliseconds{50});
```

Comment:
On different CI systems we might need to adjust the tolerance of timing.
Consider making this a configurable thing which we could leverage in other test cases (in particular metrics and async exporter).

https://github.com/census-instrumentation/opencensus-python/blob/99821a0b826445cecce0881a01790c8cc6958ceb/tests/unit/metrics/test_transport.py#L32

Reply:
I tried to pick a constant that should be large enough for most environments, though we can probably go larger without slowing down the tests too much.